File: modified_scripts/c++/matrix_multiplication/gemma_7b.cpp
```c++
int** matrix_multiplication(int** a, int** b, int m, int n, int p) {
    int** c = new int*[m];
    for (int i = 0; i < m; ++i) {
        c[i] = new int[p];
    }
    for (int i = 0; i < m; ++i) {
        for (int j = 0; j < p; ++j) {
            for (int k = 0; k < n; ++k) {
                c[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    return c;
}
```

File: modified_scripts/scala/matrix_multiplication/gemma_7b.scala

object MatrixMultiplication {

  def multiply(a: Array[Array[Int]], b: Array[Array[Int]]) : Array[Array[Int]] = {
    val result = Array.ofDim(a.length, b(0).length)

    for (i <- 0 until a.length) {
      for (j <- 0 until b(0).length) {
        for (k <- 0 until b.length) {
          result(i)(j) += a(i)(k) * b(k)(j)
        }
      }
    }

    result
  }

}
